xroad-catalog-lister/src/test/java/fi/vrk/xroad/catalog/lister/ApplicationTests.java

```diff
@@ -66,4 +67,183 @@ public void testListServices() {
 		assertEquals("MemberList size", 3, result.getMemberList().getMember().size());
 	}
 
+	@Test
+	public void testIsSoapService() {
+		IsSoapService request = new IsSoapService();
+		request.setServiceCode("testService");
+		request.setSubsystemCode("TestSubSystem");
+		IsSoapServiceResponse result = (IsSoapServiceResponse)new WebServiceTemplate(marshaller).marshalSendAndReceive(
+				"http://localhost:" + port + "/ws/IsSoapService/", request);
+		assertNotNull(result);
+		assertEquals("Is given service a SOAP service", true, result.isSoap());
+	}
+
+	@Test
+	public void testIsSoapServiceFalse() {
+		IsSoapService request = new IsSoapService();
+		request.setServiceCode("getRandom");
+		request.setSubsystemCode("TestSubSystem");
+		IsSoapServiceResponse result = (IsSoapServiceResponse)new WebServiceTemplate(marshaller).marshalSendAndReceive(
+				"http://localhost:" + port + "/ws/IsSoapService/", request);
+		assertNotNull(result);
+		assertEquals("Is given service a SOAP service", false, result.isSoap());
+	}
+
+	@Test
+	public void testIsSoapServiceException() {
+		boolean thrown = false;
+		String exceptionMessage = null;
+		try {
+			IsSoapService request = new IsSoapService();
+			request.setServiceCode("testService123");
+			request.setSubsystemCode("TestSubSystem");
+			new WebServiceTemplate(marshaller).marshalSendAndReceive(
+					"http://localhost:" + port + "/ws/IsSoapService/", request);
+		} catch (SoapFaultClientException e) {
+			thrown = true;
+			exceptionMessage = e.getMessage();
```

Comment:
Or maybe something like

@Test
public void whenExceptionThrown_thenAssertionSucceeds() {
    Exception exception = assertThrows(NumberFormatException.class, () -> {
        Integer.parseInt("1a");
    });
 
    String expectedMessage = "For input string";
    String actualMessage = exception.getMessage();
 
    assertTrue(actualMessage.contains(expectedMessage));
}

Reply:
This is a good suggestion, I have now substituted try-catch in the tests with this and it works.